Fix to multiple authorizations requests (issue #35)

dgomes opened the pull request:
This patch fixes issue #35 by substituting the external bash scripts with the native SCFramework.

Since the error messages no longer made sense (those associated with the app path), I new dialog was added  providing proper feedback to the user.

Code was also added to detect OSX version and to only try the proper interface (avoiding failed actions in the system preferences)

dgomes committed:
Initial solution in place for ProxyOn

dgomes committed:
issue #35 fixed
(also some minor refactoring)

chetan51 commented:
This is looking great! Thanks so much.

One issue I'm noticing is that if you quit the application while the proxy is on, it asks if you'd like to restore a direct connection before quitting (as it did before). But now, if you click "Yes", it doesn't actually restore the connection before quitting, and the internet connection is broken.

dgomes committed:
fixed issue with application closing and proxy conf not being restored

Fixed issue with ProxySetter object being deallocated before the thread
had finished by removing the thread altogether (there was no need for a
separate thread anyway)

dgomes commented:
Glad to be able to help, sidestep is my poor man VPN :)

So there was basically a race condition, which doesn't need to exist since there is no need for a separate thread to configure the proxy.

This last commit fixes the issue. Let me know if you find any more problems

jridgewell commented on AppController.m:
```diff
@@ -121,10 +125,8 @@ - (void)applicationDidFinishLaunching:(NSNotification *)aNotification {
-								 toTarget:self
-							   withObject:nil];
-		
+		[self turnWirelessProxyOffThread];
```
EDIT: Meant to comment the commit, not pull request. Sorry.

chetan51 replied:
Turning it off there is not necessary. But we'll have to update the state of the app after launch with the current state of the proxy.

On Aug 14, 2012, at 5:45 PM, Justin Ridgewell notifications@github.com wrote:

> In AppController.m:
> 
> > @@ -121,10 +125,8 @@ - (void)applicationDidFinishLaunching:(NSNotification *)aNotification {
> >     if (previousPID != 0) {
> >         XLog(self, @"Turning proxy off");
> > -       [NSThread detachNewThreadSelector:@selector(turnWirelessProxyOffThread)
> > -                                toTarget:self
> > 
> > ## \-                              withObject:nil];
> > -       [self turnWirelessProxyOffThread];
> >   Is turning the proxy off here necessary? Imagine
> 
> I launch the app
> I connect to my proxy
> I quit the app, clicking 'no' that I don't want to stop the proxy
> I resume the app at some later time
> I kind of expect the proxy to persist after I relaunch...
> 
> —
> Reply to this email directly or view it on GitHub.
